reference-implementation/lib/readable-stream-abstract-ops.js

```diff
@@ -159,6 +159,42 @@ export function IsReadableStream(x) {
   return true;
 }
 
+export function PutBackIntoReadableStream(stream, chunk) {
+  if (stream._state === 'closed') {
+    throw new TypeError('stream is closed');
```

Comment:
It seems the issue should be addressed by ReadableByteStream. I.e. specifying the `size` argument based on `maxSize`?